src/SetupApi/SetupApi+SP_DRVINFO_DATA.cs

```diff
@@ -0,0 +1,93 @@
+﻿// Copyright © .NET Foundation and Contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace PInvoke
+{
+    using System;
+    using System.Diagnostics.CodeAnalysis;
+    using System.Runtime.InteropServices;
+    using FileTime = System.Runtime.InteropServices.ComTypes.FILETIME;
+
+    /// <content>
+    /// Contains the <see cref="SP_DRVINFO_DATA" /> nested type.
+    /// </content>
+    public partial class SetupApi
+    {
+        /// <summary>
+        /// An <see cref="SP_DRVINFO_DATA"/> structure contains information about a driver. This structure is a member of a driver information list
+        /// that can be associated with a particular device instance or globally with a device information set.
+        /// </summary>
+        /// <seealso href="https://msdn.microsoft.com/en-us/library/windows/hardware/ff553287(v=vs.85).aspx"/>
+        [StructLayout(LayoutKind.Sequential, CharSet = CharSet.Unicode, Pack = 4)]
+        public struct SP_DRVINFO_DATA
+        {
+            /// <summary>
+            /// The size, in bytes, of the <see cref="SP_DRVINFO_DATA"/> structure.
+            /// </summary>
+            public int Size;
+
+            /// <summary>
+            /// The type of driver represented by this structure.
+            /// </summary>
+            public DriverType DriverType;
+
+            /// <summary>
+            /// Reserved. For internal use only.
+            /// </summary>
+            public UIntPtr Reserved;
+
+            /// <summary>
+            /// A <see cref="string"/> that describes the device supported by this driver.
+            /// </summary>
+            [MarshalAs(UnmanagedType.ByValTStr, SizeConst = LineLength)]
```

Comment:
Per our [contributing doc](https://github.com/dotnet/pinvoke/blob/5078a266e37139a63685362e9d9b9f795cb93973/CONTRIBUTING.md#field-types), please avoid `MarshalAs` on struct fields since this makes the structs incompatible with passing around as pointers. That doc has links to examples of what you can do instead.